### PR TITLE
Eliminate cpp featureset

### PR DESCRIPTION
**Pull request by robertsipka**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -333,23 +307,23 @@ project (Jerry CXX C ASM)
...
-     set(SOURCE_JERRY_STANDALONE_MAIN_MCU_STM32F4 main-mcu.cpp)
+     set(SOURCE_JERRY_STANDALONE_MAIN_MCU_STM32F4 main-mcu.c)
 
  # Unit tests main modules
-  file(GLOB SOURCE_UNIT_TEST_MAIN_MODULES tests/unit/*.cpp)
+  file(GLOB SOURCE_UNIT_TEST_MAIN_MODULES tests/unit/*.c)
 
 # Imported libraries
-if(("${PLATFORM}" STREQUAL "DARWIN") AND (NOT (CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)))
+if(("${PLATFORM}" STREQUAL "DARWIN") AND (NOT (CMAKE_COMPILER_IS_GNUCC)))
```
Comment (akiss77) [inline CMakeLists.txt]: Unnecessary parentheses around `CMAKE_COMPILER_IS_GNUCC`

**Comment by akiss77** [comment]
Please, also update docs/DEVELOPMENT.md

**Comment by zherczeg** [comment]
Please provide some statistics about code size and performance.

**Comment by akiss77** [comment]
`CMakeLists.txt` still contains quite some logic for `libgcc_eh.a`. With C++, and thus exception handling and stack unwinding gone, is `libgcc_eh.a` still needed?

(The `libclang_rt.osx.a` is also suspicious.)

File: jerry-core/ecma/builtin-objects/ecma-builtin-helpers.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2015 Samsung Electronics Co., Ltd.
+ /* Copyright 2015 Samsung Electronics Co., Ltd.
```
Comment (LaszloLango) [inline jerry-core/ecma/builtin-objects/ecma-builtin-helpers.h]: accident change, remove space

**Comment by robertsipka** [comment]
Thanks for the comments, I've updated the patch and rebased to master.

File: docs/DEVELOPMENT.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 Currently, only Ubuntu 14.04+ is officially supported as primary development environment.
 
 There are several dependencies, that should be installed manually. The following list is required for building:
-- `gcc` or `g++` higher than `4.8.2`
+- `gcc` higher than `4.8.2`
```
Comment (zherczeg) [inline docs/DEVELOPMENT.md]: Do we need gcc 4.8.2 for a C99 compatible engine?

**Comment by zherczeg** [comment]
Nice job! LGTM. I think we could reduce the minimum GCC version, but that is not important at the moment.

**Pull request merged**